lib/semantics/check-omp-structure.cc

```diff
@@ -220,11 +246,28 @@ void OmpStructureChecker::Enter(const parser::OmpClause::NumTasks &) {
 void OmpStructureChecker::Enter(const parser::OmpClause::NumTeams &) {
   CheckAllowed(OmpClause::NUM_TEAMS);
 }
-void OmpStructureChecker::Enter(const parser::OmpClause::NumThreads &) {
+void OmpStructureChecker::Enter(const parser::OmpClause::NumThreads &x) {
   CheckAllowed(OmpClause::NUM_THREADS);
+  const auto v{GetIntValue(x.v)};
+  if (v && v.value() <= 0) {
+    context_.Say(GetContext().clauseSource,
+        "The parameter of the NUM_THREADS clause must be "
+        "a positive integer expression"_err_en_US);
+  }
+  // if parameter is variable, defer to Expression Analysis
 }
-void OmpStructureChecker::Enter(const parser::OmpClause::Ordered &) {
+
+void OmpStructureChecker::Enter(const parser::OmpClause::Ordered &x) {
   CheckAllowed(OmpClause::ORDERED);
+  // the parameter of ordered clause is optional
+  if (const auto &expr{x.v}) {
+    const auto v{GetIntValue(expr)};
```

Comment:
same here

Reply:
I made all the changes you suggested, thanks!